src/vmm/src/resources.rs

```diff
@@ -153,6 +159,67 @@ impl VmResources {
         Ok(resources)
     }
 
+    /// Configures VmResources from the state of the Vmm.
+    pub fn from_vmm(vmm: &Vmm) -> std::result::Result<Self, Error> {
+        let mut resources = Self::default();
+        resources.boot_config = None;
+
+        if let Ok(balloon_config) = vmm.balloon_config() {
+            resources
+                .set_balloon_device(BalloonDeviceConfig::from(balloon_config))
+                .map_err(Error::BalloonDevice)?;
+        }
+
+        vmm.mmio_device_manager.for_each_virtio_device(|virtio_type, _device_id, _device_info, virtio_device| {
```

Comment:
We can limit the calls to `for_each_virtio_device` to just one, by `match`-ing on the `virtio_type` inside the closure

Reply:
That makes perfect sense. I'll definitely do that :+1: